WordPress/gutenberg
# Update background image control dropdown height to fit-content

Fixes: #68029 

## What?
This PR contains a patch for fixing spacing inconsistency in the background image selector button.

## How?
As the issue description mentions, the default size of `Toolbar Button` was updated to `compact`, whose styles are applied using the `is-compact` classname.

The `is-compact` classname contains a height of `32px` as mentioned here:
https://github.com/WordPress/gutenberg/blob/581de9f893ff63c156b118f7244211f417fe7543/packages/components/src/button/style.scss#L298

https://github.com/WordPress/gutenberg/blob/581de9f893ff63c156b118f7244211f417fe7543/packages/base-styles/_variables.scss#L99

Whereas, the heights described in `.components-dropdown` applied to its parent, is actually `36px` leading to a mismatch in the parent container and child's heights.

https://github.com/WordPress/gutenberg/blob/581de9f893ff63c156b118f7244211f417fe7543/packages/block-editor/src/components/background-image-control/style.scss#L26

https://github.com/WordPress/gutenberg/blob/581de9f893ff63c156b118f7244211f417fe7543/packages/block-editor/src/components/background-image-control/style.scss#L47

To fix the issue, the heights are updated inside `.components-dropdown` to now use `fit-content` instead of using the previously hardcoded value `36px`.

As an alternative, we can also hardcode the height to `32px`.

## File Changed:
`> packages/block-editor/src/components/background-image-control/style.scss`

## Testing Instructions
1. Go to the site editor.
2. Click on "Styles"
3. Click on "Background"
4. Confirm that the spacing for the button is now correct.

## Screenshots

|Before|After|
|-|-|
|![Screenshot 2024-12-17 at 10 38 17 AM](https://github.com/user-attachments/assets/48e3b41a-6463-4210-849a-2990a4c79c4e)|![Screenshot 2024-12-17 at 10 37 24 AM](https://github.com/user-attachments/assets/d6bcc79c-3409-4f08-8b1b-af9ee51ae03b)|
